src/vm/dllimport.cpp

```diff
@@ -7047,7 +7055,6 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
     bool libNameIsRelativePath = Path::IsRelative(wszLibName);
     if (hmod == NULL)
     {
-#ifndef FEATURE_CORECLR
         // First checks if the method has DefaultDllImportSearchPathsAttribute. If method has the attribute
```

Comment:
The order of these lookups are very fragile. If you move the host path lookup earlier, it might introduce some other breaking changes since other DLL under nuget package directory might get chosen over the one along side the assembly, which is incorrect. On the other hand, you do want the service path to have highest priority over everything else. We might need to have a separate discussion on the servicing scenario.